chessence-frontend/src/pages/Archive/GameList/GameList.js

```diff
@@ -21,24 +21,27 @@ export default function GameList(props) {
         }
     }
 
-    useEffect(() => {
-        const fetchUsernames = async () => {
-            const usernamePromises = props.gameData.map(async (row) => {
-                const black = await getUserName(row.whiteID);
-                const white = await getUserName(row.blackID);
-                const winner = await getUserName(row.winner);
-                return { black, white, winner };
-            });
-
-            const resolvedUsernames = await Promise.all(usernamePromises);
-            setUserNames(resolvedUsernames);
-        };
+    async function fetchUsernames() {
+        const users = [];
+    
+        for (const row of props.gameData) {
+            const black = await getUserName(row.whiteID);
```

Comment:
Should this be "const white = await getUserName(row.whiteID);"?
And then the line below, "const white = await getUserName(row.blackID);"?

Reply:
You are right! My bad